### PR TITLE
Docs resgroups diskio

### PR DESCRIPTION
**Pull request by mperezfuster**
This PR documents the new limit for resource groups io_limit, which limits disk I/O.
There are changes to the narrative pages, the CREATE/ALTER RESOURCE GROUP reference pages, and the relevant gp_toolkit views.
Note there is another PR https://github.com/greenplum-db/gpdb/pull/16204 that is already addressing some changes for the narrative page Using Resource Groups. I have not included those changes in this PR.

@SmartKeyerror feel free to add other reviewers to this PR.

Review sites:

Using Resource Groups: https://docs-staging.vmware.com/en/draft/VMware-Tanzu-Greenplum/mireia-resgroups/greenplum-database/admin_guide-workload_mgmt_resgroups.html

System views: https://docs-staging.vmware.com/en/draft/VMware-Tanzu-Greenplum/mireia-resgroups/greenplum-database/ref_guide-system_catalogs-catalog_ref-views.html#gp_resgroup_iostats_per_host

SQL Commands: 
https://docs-staging.vmware.com/en/draft/VMware-Tanzu-Greenplum/mireia-resgroups/greenplum-database/ref_guide-sql_commands-CREATE_RESOURCE_GROUP.html
https://docs-staging.vmware.com/en/draft/VMware-Tanzu-Greenplum/mireia-resgroups/greenplum-database/ref_guide-sql_commands-ALTER_RESOURCE_GROUP.html

### CODE REVIEW

File: gpdb-doc/markdown/admin_guide/workload_mgmt_resgroups.html.md [hunk 1]
```diff
@@ -30,6 +30,7 @@ When you create a resource group, you provide a set of limits that determine the
 |CPU_MAX_PERCENT|The maximum percentage of CPU resources the group can use.| [0-100] | -1 (not set)|
 |CPU_WEIGHT|The scheduling priority of the resource group.| [1-500] | 100 |
 |CPUSET|The specific CPU logical core (or logical thread in hyperthreading) reserved for this resource group.| It depends on system core configuration | -1 |
+|IO_LIMIT| The limit for the maximum read/write sequential disk I/O throughput, and maximum read/write I/O operations per second. Set the value on a per-tablespace basis.| [0-MAX] | -1 |
```
Comment (VMWareJennyBeth) [inline gpdb-doc/markdown/admin_guide/workload_mgmt_resgroups.html.md:33]: My understanding is that they can also set Disk I/O across all tablespaces in addition to per tablespace but I could totally be wrong.

File: gpdb-doc/markdown/admin_guide/workload_mgmt_resgroups.html.md [hunk 2]
```diff
@@ -149,11 +150,25 @@ There are some special usage considerations regarding memory limits:
 - The maximum value of `statement_mem` is capped at [max_statement_mem](../ref_guide/config_params/guc-list.html#max_statement_mem).
 - Queries whose plan cost is less than the limit `MIN_COST` use a memory quota of `statement_mem`. 
 
+### <a id="diskio"></a>Disk I/O Limits
+
+Greenplum Database leverages Linux control groups to implement disk I/O limits. The parameter `IO_LIMIT` limits the maximum read/write sequential disk I/O throughput, and the maximum read/write I/O operations per second for the queries assigned to a specific resource group. It allocates bandwidth, ensures the use of high-priority resource groups, and avoids excessive use of disk bandwidth. The value of the parameter is set on a per-tablespace basis.
```
Comment (VMWareJennyBeth) [inline gpdb-doc/markdown/admin_guide/workload_mgmt_resgroups.html.md:155]: Same comment as above about also being able to set for all tablespaces.

File: gpdb-doc/markdown/admin_guide/workload_mgmt_resgroups.html.md [hunk 3]
```diff
@@ -305,19 +320,21 @@ The default resource groups `admin_group`, `default_group`, and `system_group`
...
 |CPU_WEIGHT|100|100|100|
 |CPUSET|-1|-1|-1|
+|IO_LIMIT|-1|-1|-1|
 |MEMORY\_LIMIT|-1|-1|-1|
 |MIN_COST|0|0|0|
 
 ## <a id="topic10"></a>Creating Resource Groups 
 
-When you create a resource group for a role, you provide a name and a CPU resource allocation mode (core or percentage). You can optionally provide a concurrent transaction limit, a memory limit, a CPU soft priority, and a minimum cost. Use the [CREATE RESOURCE GROUP](../ref_guide/sql_commands/CREATE_RESOURCE_GROUP.html) command to create a new resource group.
+When you create a resource group for a role, you provide a name and a CPU resource allocation mode (core or percentage). You can optionally provide a concurrent transaction limit, a memory limit, a CPU soft priority, disk I/O limits,and a minimum cost. Use the [CREATE RESOURCE GROUP](../ref_guide/sql_commands/CREATE_RESOURCE_GROUP.html) command to create a new resource group.
```
Comment (VMWareJennyBeth) [inline gpdb-doc/markdown/admin_guide/workload_mgmt_resgroups.html.md:329]: missing space after "limits,"

File: gpdb-doc/markdown/ref_guide/sql_commands/CREATE_RESOURCE_GROUP.html.md [hunk 4]
```diff
@@ -11,31 +11,28 @@ CREATE RESOURCE GROUP <name> WITH (<group_attribute>=<value> [, ... ])
...
+Creates a new resource group for Greenplum Database resource management. 
 
-A resource group that you create to manage a user role identifies concurrent transaction, memory, and CPU limits for the role when resource groups are enabled. You may assign such resource groups to one or more roles.
-
-A resource group that you create to manage the resources of a Greenplum Database external component such as PL/Container identifies the memory and CPU limits for the component when resource groups are enabled. These resource groups use cgroups for both CPU and memory management. Assignment of resource groups to external components is component-specific. For example, you assign a PL/Container resource group when you configure a PL/Container runtime. You cannot assign a resource group that you create for external components to a role, nor can you assign a resource group that you create for roles to an external component.
+A resource group that you create to manage a user role identifies concurrent transaction, memory, CPU and disk I/O limits for the role when resource groups are enabled. You may assign such resource groups to one or more roles.
```
Comment (VMWareJennyBeth) [inline gpdb-doc/markdown/ref_guide/sql_commands/CREATE_RESOURCE_GROUP.html.md:26]: We use the serial comma. So add a comma after CPU in that list.

File: gpdb-doc/markdown/ref_guide/sql_commands/CREATE_RESOURCE_GROUP.html.md [hunk 5]
```diff
@@ -64,6 +61,18 @@ CPUSET <coordinator_cores>;<segment_cores>
 
 :   > **Note** You can configure `CPUSET` for a resource group only after you have enabled resource group-based resource management for your Greenplum Database cluster.
 
+IO_LIMIT '<tablespace_name>:wbps=<wbps_value>, rbps=<rbps_value>, wiops=<wiops_value>, riops=<riops_value>'
+
+:   Optional. The maximum read/write sequential disk I/O throughput, and the maximum read/write I/O operations per second for the queries assigned to a specific tablespace and resource group. When you use this parameter, you may speficy:
```
Comment (VMWareJennyBeth) [inline gpdb-doc/markdown/ref_guide/sql_commands/CREATE_RESOURCE_GROUP.html.md:66]: speficy => specify

Also, same comment about setting Disk I/O for all tablespaces.

File: gpdb-doc/markdown/admin_guide/workload_mgmt_resgroups.html.md (hunk 1 above)
Comment (mperezfuster) [reply gpdb-doc/markdown/admin_guide/workload_mgmt_resgroups.html.md:33]: @SmartKeyerror can you confirm if we can set the disk io limits without specifying a tablespace?

File: gpdb-doc/markdown/admin_guide/wlmgmt.html.md [hunk 6]
```diff
@@ -15,7 +15,7 @@ The following table summarizes some of the differences between resource queues a
 |Concurrency|Managed at the query level|Managed at the transaction level|
 |CPU|Specify query priority|Specify percentage of CPU resources or the specific number of CPU cores; uses Linux Control Groups|
 |Memory|Managed at the queue and operator level; users can over-subscribe|Managed at the transaction level, with enhanced allocation and tracking; users can over-subscribe|
-|Memory Isolation|None|None|
+|Disk I/O|None|Limit the maximum read/write sequential disk I/O throughput, and maximum read/write I/O operations per second|
```
Comment (RMTT) [inline gpdb-doc/markdown/admin_guide/wlmgmt.html.md:18]: diskio is not specific for `sequential` io , so i think `sequential` should be removed here.

File: gpdb-doc/markdown/admin_guide/workload_mgmt_resgroups.html.md (hunk 1 above)
Comment (RMTT) [reply gpdb-doc/markdown/admin_guide/workload_mgmt_resgroups.html.md:33]: disk io limits must start with a tablespace, but you can use `*` as tablespace name which stand all tablespaces.

btw, `sequential` should be removed.

File: gpdb-doc/markdown/admin_guide/workload_mgmt_resgroups.html.md (hunk 2 above)
Comment (RMTT) [reply gpdb-doc/markdown/admin_guide/workload_mgmt_resgroups.html.md:155]: `sequential` should be removed.

File: gpdb-doc/markdown/ref_guide/sql_commands/CREATE_RESOURCE_GROUP.html.md [hunk 7]
```diff
@@ -11,31 +11,28 @@ CREATE RESOURCE GROUP <name> WITH (<group_attribute>=<value> [, ... ])
 where group_attribute is one of:
 
 ```
+[ CONCURRENCY=<integer> ]
 CPU_MAX_PERCENT=<integer> | CPUSET=<coordinator_cores>;<segment_cores>
-[ MEMORY_LIMIT=<integer> ]
 [ CPU_WEIGHT=<integer> ]
-[ CONCURRENCY=<integer> ]
+[IO_LIMIT='<tablespace_name>:wbps=<wbps_value>, rbps=<rbps_value>, wiops=<wiops_value>, riops=<riops_value>']
```
Comment (RMTT) [inline gpdb-doc/markdown/ref_guide/sql_commands/CREATE_RESOURCE_GROUP.html.md:17]: multiple tablespaces use `;` as separator, for example:

```
[IO_LIMIT='<tablespace_name>:wbps=<wbps_value>, rbps=<rbps_value>, wiops=<wiops_value>, riops=<riops_value>;<tablespace_name2>:wbps=<wbps_value>, rbps=<rbps_value>, wiops=<wiops_value>, riops=<riops_value>']
```

File: gpdb-doc/markdown/ref_guide/sql_commands/CREATE_RESOURCE_GROUP.html.md [hunk 8]
```diff
@@ -64,6 +61,18 @@ CPUSET <coordinator_cores>;<segment_cores>
 
 :   > **Note** You can configure `CPUSET` for a resource group only after you have enabled resource group-based resource management for your Greenplum Database cluster.
 
+IO_LIMIT '<tablespace_name>:wbps=<wbps_value>, rbps=<rbps_value>, wiops=<wiops_value>, riops=<riops_value>'
+
+:   Optional. The maximum read/write sequential disk I/O throughput, and the maximum read/write I/O operations per second for the queries assigned to a specific tablespace and resource group. When you use this parameter, you may specify:
```
Comment (RMTT) [inline gpdb-doc/markdown/ref_guide/sql_commands/CREATE_RESOURCE_GROUP.html.md:66]: `sequential` should be removed.

File: gpdb-doc/markdown/ref_guide/system_catalogs/catalog_ref-views.html.md [hunk 9]
```diff
@@ -281,11 +282,43 @@ The `gp_toolkit.gp_resgroup_config` view allows administrators to see the curren
...
+|`rbps`|bigint||The real-time read sequential disk I/O throughput by the resource group on a host, in MB/S|
+|`wbps`|bigint||The real-time write sequential disk I/O throughput by the resource group on a host, in MB/S|
+|`riops`|bigint||The real-time read I/O operations per second by the resource group on a host|
+|`wiops`|bigint||The real-time write I/O operations per second by the resource group on a host|
+
+Sample output for the `gp_resgroup_iostats_per_host` view:
+
+```
+SELECT * from gp_toolkit.gp_resgroup_iostats_per_host;
+ rsgname        | hostname | tablespace       | rbps (MB_read/s) | wbps (MB_wrtn/s) | riops (r/s) | wiops (w/s)
```
Comment (RMTT) [inline gpdb-doc/markdown/ref_guide/system_catalogs/catalog_ref-views.html.md:313]: The column names have  changed:

`rbps (MB_read/s)` to `rbps`
`wbps (MB_wrtn/s)` to `wbps`
`riops (r/s)` to `riops`
`wiops (w/s)` to `wiops`

**Review by RMTT — CHANGES_REQUESTED** [review]

File: gpdb-doc/markdown/admin_guide/workload_mgmt_resgroups.html.md (hunk 2 above)
Comment (deenav) [inline gpdb-doc/markdown/admin_guide/workload_mgmt_resgroups.html.md:155]: We should explicitly say that IO_LIMIT is supported only on `cgroup-v2`

The below warning will appear if user try to configure with 'cgroup' v1
```
# create resource group rg2 WITH (cpu_max_percent=20, io_limit='myspace:rbps=5,riops=1000');
WARNING:  resource group io limit only can be used in cgroup v2.  (seg5 192.168.121.4:6003 pid=127882)
```

File: gpdb-doc/markdown/admin_guide/workload_mgmt_resgroups.html.md [hunk 10]
```diff
@@ -30,6 +30,7 @@ When you create a resource group, you provide a set of limits that determine the
 |CPU_MAX_PERCENT|The maximum percentage of CPU resources the group can use.| [0-100] | -1 (not set)|
 |CPU_WEIGHT|The scheduling priority of the resource group.| [1-500] | 100 |
 |CPUSET|The specific CPU logical core (or logical thread in hyperthreading) reserved for this resource group.| It depends on system core configuration | -1 |
+|IO_LIMIT| The limit for the maximum read/write disk I/O throughput, and maximum read/write I/O operations per second. Set the value on a per-tablespace basis.| [0-MAX] | -1 |
```
Comment (RMTT) [inline gpdb-doc/markdown/admin_guide/workload_mgmt_resgroups.html.md:33]: The value range should be `[2, 4294967295]` or `MAX`.

**Review by RMTT — CHANGES_REQUESTED** [review]

**Pull request closed without merging**